gufe/mapping/atom_mapping_scorer.py

```diff
@@ -0,0 +1,42 @@
+# This code is part of kartograf and is licensed under the MIT license.
+# For details, see https://github.com/OpenFreeEnergy/gufe
+
+import abc
+
+from . import AtomMapping
+
+
+class AtomMappingScorer(abc.ABC):
+    """A generic class for scoring Atom mappings.
+    this class can be used for example to build graph algorithm based networks.
+
+    Implementations of this class can require an arbitrary and non-standardised
+    number of input arguments to create.
+
+    Implementations of this class provide the :meth:`.get_score` method
+
+    """
+
+    def __call__(self, mapping: AtomMapping, *args, **kwargs) -> float:
+        return self.get_score(mapping)
+
+    @abc.abstractmethod
+    def get_score(self, mapping: AtomMapping, *args, **kwargs) -> float:
```

Comment:
rather than allow args/kwargs to this, could this instead be done by having all args/kwargs passed to the init?

so:
``` python
scorer = MyScorer()
score = scorer(mapping, alpha=0.1)

# instead
scorer = MyScorer(alpha=0.1)
score = scorer(mapping)
```
I think the latter is better, unless there's some use case that it doesn't allow?

Reply:
aehm, yes you are very right here!